common/masker/masker.go

```
@@ -0,0 +1,107 @@
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package masker
+
+import (
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+)
+
+const passwordMask = "******"
+
+var (
+	DefaultFieldNames     = []string{"Password", "KeyData"}
+	DefaultYAMLFieldNames = []string{"password", "keyData"}
+)
+
+// MaskYaml replace password values with mask and returns copy of the string.
+// Does recursive replacement for entire yamlStr.
+func MaskYaml(yamlStr string, fieldNamesToMask []string) string {
+	fns := make(map[string]struct{}, len(fieldNamesToMask))
+	for _, fieldName := range fieldNamesToMask {
+		fns[fieldName] = struct{}{}
+	}
+
+	var parsedYaml map[string]interface{}
+	err := yaml.Unmarshal([]byte(yamlStr), &parsedYaml)
+	if err != nil {
+		return yamlStr
+	}
+
+	maskMap(parsedYaml, fns)
+
+	strBytes, err := yaml.Marshal(parsedYaml)
+	if err != nil {
+		return yamlStr
```

same here, try not to ignore errors